sdk/spring/spring-cloud-azure-core/src/main/java/com/azure/spring/core/converter/AzureHttpProxyOptionsConverter.java

```diff
@@ -15,33 +16,37 @@
 /**
  * Converts a {@link HttpProxyProperties} to a {@link ProxyOptions}.
  */
-public final class AzureHttpProxyOptionsConverter implements Converter<HttpProxyProperties, ProxyOptions> {
+public final class AzureHttpProxyOptionsConverter implements Converter<ProxyAware.Proxy, ProxyOptions> {
 
     private static final Logger LOGGER = LoggerFactory.getLogger(AzureHttpProxyOptionsConverter.class);
     public static final AzureHttpProxyOptionsConverter HTTP_PROXY_CONVERTER = new AzureHttpProxyOptionsConverter();
 
     @Override
-    public ProxyOptions convert(HttpProxyProperties proxyProperties) {
-        if (!StringUtils.hasText(proxyProperties.getHostname()) || proxyProperties.getPort() == null) {
+    public ProxyOptions convert(ProxyAware.Proxy proxy) {
+        if (!StringUtils.hasText(proxy.getHostname()) || proxy.getPort() == null) {
             LOGGER.debug("Proxy hostname or port is not set.");
             return null;
         }
 
-        final String type = proxyProperties.getType();
+        final String type = proxy.getType();
         ProxyOptions.Type sdkProxyType;
         if ("http".equalsIgnoreCase(type)) {
             sdkProxyType = ProxyOptions.Type.HTTP;
         } else {
             sdkProxyType = ProxyOptions.Type.SOCKS4;
         }
 
-        ProxyOptions proxyOptions = new ProxyOptions(sdkProxyType, new InetSocketAddress(proxyProperties.getHostname(),
-                                                                                         proxyProperties.getPort()));
-        if (StringUtils.hasText(proxyProperties.getUsername()) && StringUtils.hasText(proxyProperties.getPassword())) {
-            proxyOptions.setCredentials(proxyProperties.getUsername(), proxyProperties.getPassword());
+        ProxyOptions proxyOptions = new ProxyOptions(sdkProxyType, new InetSocketAddress(proxy.getHostname(),
+                                                                                         proxy.getPort()));
+        if (StringUtils.hasText(proxy.getUsername()) && StringUtils.hasText(proxy.getPassword())) {
+            proxyOptions.setCredentials(proxy.getUsername(), proxy.getPassword());
         }
-        if (StringUtils.hasText(proxyProperties.getNonProxyHosts())) {
-            proxyOptions.setNonProxyHosts(proxyProperties.getNonProxyHosts());
+
+        if (proxy instanceof HttpProxyProperties) {
```

Comment:
instance of httpproxy interface instead of type?

Reply:
Yes, the global properties will also use the same proxy converter, even though it's not the proxy type of HTTP.

Reply:
I mean we should use the interface `ProxyAware.HttpProxy` here instead of `HttpProxyProperties`